### PR TITLE
Migrate onnx-genai metadata emitters to the current pipeline/speculative contract

### PR DESCRIPTION
**Pull request by justinchuby**
Fixes six long-standing schema-conformance failures caused by mobius's legacy onnx-genai emitters drifting from the runtime's published JSON schema.

## Why the drift was invisible

The conformance tests searched a couple of hard-coded local onnx-genai checkout paths and `pytest.skip`ped when none was found. CI has no such checkout, so **six tests never ran there**; on the machines that did have one, the result depended on whatever revision that clone happened to sit on. Two upstream contract redesigns accumulated unnoticed.

## The two contract redesigns

1. **`pipeline` is now `PipelineSpec`** — a single property, `workflow` (a typed SSA graph), with `additionalProperties: false`. The legacy emitters produced `{models, dataflow, strategy, phases}`, all of which the schema now rejects.
2. **`speculative` is now `SpeculativeContract`** — requires `{proposer, target, vocabulary, max_proposal_width}` and forbids every field of the old flat MTP block. The flat `SpeculatorConfig` that block was modelled on still exists upstream, but it describes a HuggingFace `config.json` section, not `InferenceMetadata.speculative`.

### Diffusion — `build_diffusion_pipeline_metadata`
Now emits the denoise loop explicitly: the solver, sigma/alpha schedule, timestep table, guidance combine and output clamp are real ONNX components built from mobius's policy library and shipped with the document.

It **cannot** delegate to `build_diffusion_workflow_metadata`: the ComfyUI conversion path has no component graphs at all (it deliberately does not build or export them), and that builder derives everything from live `ir.Model`s. So the workflow is built directly, reusing `workflow_metadata`'s `_invoke` / `_publish_workflow_v1` and `mobius.generation`'s solver builders. The schedule is derived from the scheduler's own betas through the same diffusers-compatible helpers the package exporter uses, so a ComfyUI conversion and a package export of the same checkpoint describe the same dynamics. img2img's `start_step` lowers to a sliced schedule; the VAE `scaling_factor` and the sigma-space initial-state scale are emitted as explicit components.

Three cases now **fail closed** instead of being silently mis-described:
- an ancestral sampler (mobius ships no stochastic solver),
- Karras/exponential sigma spacing (the workflow ships the sigma table as a constant, so a `use_karras_sigmas` hint is no longer sufficient),
- a latent-only graph with no VAE decode.

### MTP — `write_mtp_speculator_metadata`
Emits a `SpeculativeContract` anchored to the backbone's workflow: it registers the head as a workflow component, names the target by its declared `logits` port role, states the hidden handoff as `port_bindings.target_hidden_context` plus a `hidden_states` role on the target output, and completes the rollback capacity its own claim requires.

It is declared `block`, not `chained`: a chained proposer must expose a `logits_output` carrying the next-token distribution, and this sidecar emits only `mtp_hidden` — the runtime decodes it through the target's shared LM head, which is why that initializer is listed in `shared_weights`.

### VLM
`build_native_vlm_package_metadata` produces mobius's **internal** structural descriptor, not a publishable document — `build_vlm_workflow_metadata` already consumes it and republishes only `preprocessing` under a real `pipeline.workflow`. The actual bug was that `write_native_vlm_package_metadata` wrote that descriptor to `inference_metadata.yaml`. It now writes the workflow document, and the tests validate the published document instead of the descriptor.

### CI visibility
The upstream schema is vendored under `src/mobius/integrations/onnx_genai/_schema/` and is the default, so conformance never skips and drift becomes a test failure. A local checkout is no longer consulted implicitly — one that is ahead of or behind `main` reintroduces exactly the machine-dependent result that hid this bug. Set `ONNX_GENAI_SCHEMA` to validate against a specific revision.

Three further conformance tests (codec, speech-to-text, duplex workflows) were skipping for the same reason and now run.

## Tests

Migrated in the same change rather than loosened:
- `TestMtpSpeculatorMetadata::test_exact_schema_keys_and_values` asserts the new contract; `test_no_legacy_field_names` is repointed at the now-legacy field names.
- A new MTP test anchors against a workflow `write_decoder_workflow_metadata` actually emits (which ships ~11 generated policy components), so target selection can't regress to "the only ONNX component".
- New tests pin that the solver schedule comes from the scheduler's betas rather than a placeholder ramp.
- Diffusion/ComfyUI tests assert workflow structure instead of the removed `strategy` block; scheduler facts that were duplicated into the document are asserted on the parsed `ComfyUIWorkflow` where they actually live.

## Verification

Rebased onto current `main` (includes #552 and #553).

```
python -m pytest tests/build_graph_test.py tests/cli_test.py src/ -q \
  -k "not phi4mm and not apply_weights_unknown" -n auto
```

| | before (`main` at branch point) | after |
|---|---|---|
| full suite | 8 failed / 4613 passed | **0 failed / 4641 passed** |
| `src/mobius/integrations/onnx_genai/` with `ONNX_GENAI_SCHEMA` at upstream `main` | 7 failed / 240 passed | **258 passed / 1 skipped** |

Six of those eight failures are the schema-drift ones fixed here. The other two — `qwen_image_test.py::test_deterministic_l4_l5_image_edit_golden` and `TestNativeVlmPackageMetadata::test_cached_gemma_processor_matches_emitted_patch_budget` — were fixed by #553, which is now merged; this branch preserves both of its changes through the rebase.

`lintrunner` is clean.

## Sequencing note

`workflow_metadata.py` and `auto_export.py` are **not modified** (both are read from only), so this does not conflict with #551.

### CODE REVIEW

File: src/mobius/integrations/onnx_genai/comfyui.py [hunk 1]
```diff
@@ -318,13 +318,22 @@ def parse_comfyui_workflow(
...
+            f"image output, and there is nothing to decode with; supported decode nodes: "
+            f"{', '.join(sorted(_VAE_DECODE_NODES))}. How to fix: add a VAEDecode node to "
+            "the workflow before converting it."
+        )
+
     metadata = build_diffusion_pipeline_metadata(
         num_inference_steps=steps,
         scheduler=sched,
         guidance_scale=guidance,
         start_step=start_step or None,
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply src/mobius/integrations/onnx_genai/comfyui.py:358]: Good catch — this was a real gap. Fixed in a52e72c.

`parse_comfyui_workflow` was building the sampler graphs and discarding them, so a caller holding only the metadata dict had no way to produce the `policies/*.onnx` the document declares. `ComfyUIWorkflow` now carries them in `policy_components` and exposes `save_policy_components(directory)`; the dict-only wrappers (`translate_comfyui_workflow` / `..._file`) document that they drop the graphs and point at either `convert_comfyui_workflow` or the parse result.

Two tests cover it: one asserts every `policies/*` artifact the parse result references is materializable, the other that `convert_comfyui_workflow` writes exactly the referenced set — the equality direction also catches stale artifacts.

While here I left `convert_comfyui_workflow` building its own package rather than reusing the parse-time one, and said so in a comment: the checkpoint's `scheduler_config` can reconcile to a different solver than the ComfyUI sampler implied (e.g. Euler → DDIM), and reusing the parse-time package would leave that run's `model_input_scale` behind for a document that never declares it. The written-set-equals-referenced-set assertion is what guards that if someone later "simplifies" it.

File: src/mobius/integrations/onnx_genai/inference_metadata.py [hunk 2]
```diff
@@ -2319,115 +2429,467 @@ def build_diffusion_pipeline_metadata(
...
-            latent (``denoiser_sample_input``).
-        vae_latent_input: VAE latent input port name.
-        text_encoder_filename: Optional text encoder; runs ``prompt_only`` and
-            feeds ``denoiser_conditioning_input``.
-        text_encoder_output: Text encoder output port name.
+        scheduler: Noise-schedule config (defaults to DDIM defaults). Its
+            ``kind`` selects the solver component.
+        timesteps: Explicit per-step timestep table; a linear ramp otherwise.
+        schedule: Explicit sigma schedule (``num_inference_steps + 1`` values);
+            a linear ramp otherwise.
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply src/mobius/integrations/onnx_genai/inference_metadata.py]: Agreed, and this was already fixed in cd1f69e (rebased to dbcc488) — the review ran against the earlier revision of the branch.

The Args now read:

```
timesteps: Explicit per-step timestep table; derived from ``scheduler``
    otherwise.
schedule: Explicit solver schedule (``num_inference_steps + 1`` values);
    derived from ``scheduler`` otherwise.
```

`grep -rn "linear ramp" src/mobius/` returns no hits in this module. The "linear ramp" wording described a genuinely wrong earlier implementation, not just a stale doc: the schedule is what `solver_step` integrates (cumulative alphas for DDIM, sigmas for Euler / DPM-Solver++), so a placeholder ramp would have denoised along the wrong trajectory. `test_schedule_is_derived_from_the_scheduler_not_invented` now pins the emitted constant against `_ddim_alpha_schedule` and explicitly asserts it is *not* the `1 - i/n` ramp.
